tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/IRChecker.scala

```diff
@@ -1060,9 +1060,16 @@ class IRChecker(unit: LinkingUnit, logger: Logger) {
 }
 
 object IRChecker {
-  private final class ErrorContext private (pos: Position, name: String) {
-    override def toString(): String =
+  private final class ErrorContext private (val treeOrLinkedClass: Any)
+      extends AnyVal {
+
+    override def toString(): String = {
+      val (pos, name) = treeOrLinkedClass match {
```

Comment:
Not necessarily. It doesn't matter at this point, because `toString` is never called in practice. It's called only if there is an IR checker error, in which case we can be slow and memory intensive and what not.